src/modules/cache-client/cache-client.service.ts

```diff
@@ -598,20 +602,37 @@ export class CacheClient implements ICacheClient {
       return true;
     }
 
-    if (!config || !this.authEnabled || !config.url) {
-      return false;
+    // Retry server errors
+    if (response.status >= 500) {
+      return true;
     }
 
-    const isAuthError = response.status === 401 || response.status === 403;
+    const isAuthError = [401, 403, 407].includes(response.status);
+    const clientErrorsToRetry = [401, 403, 407, 408, 411, 412, 425, 426];
+
     const isAuthEndpoint =
-      config.url?.indexOf('/login') >= 0 &&
-      config.url?.indexOf('/refresh_token') >= 0 &&
-      config.url?.indexOf(TEST_LOGIN_ENDPOINT) >= 0;
+      config.url &&
+      (config.url.indexOf('/login') >= 0 ||
+        config.url.indexOf('/refresh_token') >= 0 ||
+        config.url.indexOf(TEST_LOGIN_ENDPOINT) >= 0);
 
-    if (!isAuthError || isAuthEndpoint) {
+    if (isAuthEndpoint) {
       return false;
     }
 
+    // Retry some client errors
+    if (
```

Comment:
Is this the same as `if (clientErrorsToRetry.includes(...`? Also we can check this without introducing new var similarly as isAuthError

Reply:
I know, but i think this is more readable, if you read the var name and you know what is going on instead of  analyzing the code